### PR TITLE
GitHub actions: upgrade `setup-node` and `checkout` from v1/2 to v3

### PR DESCRIPTION
**Pull request by troley**
Ref. semestry/tasks#204

## Proposed changes

See title

## Testing

N/A

## Documentation

N/A

### CODE REVIEW

**Comment by johanwortel** [comment]
My trigger was `node12` in `action.yml`.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,10 +14,10 @@ jobs:
...
     steps:
       - name: Checkout Repo
-        uses: actions/checkout@v2.0.0
+        uses: actions/checkout@v3
 
       - name: Install Node
-        uses: actions/setup-node@v1.3.0
+        uses: actions/setup-node@v3
         with:
           node-version: "12.x"
```
[the remark this answers is not in the document]
Comment (troley) [reply .github/workflows/ci.yml:23]: Yes, I fixed that one right away as well.

**Comment by troley** [comment]
I'm clueless what's failing here. I think it might be the ESLint task in `ci.yml`. It's using an `xt0rted/eslint-action` commit which is from an outdated archived repository. Maybe removing the ESLint task would help or else I suggest we leave this as it is.

**Comment by johanwortel** [comment]
The master version seems incomplete to me:
- `./dist/index.js` is the file to run with node. This is a bundle, but I cannot find when and how it is created.
  - Should `./main.js` be used without bundling?
- Something to do with the earlier switch from Docker to Node?
- If the bundle is present, eslint should not check it. It is an output artifact based on external packages. Does that need a configuration option?

**Comment by MikeN123** [comment]
lol, probably I never documented how that dist bundle was created...

**Comment by MikeN123** [comment]
The precompiling is explained here: https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action#commit-tag-and-push-your-action-to-github

Basically, something like:

```
npm i -g @vercel/ncc
ncc build main.js
```

I think...

**Comment by johanwortel** [comment]
IntelliJ IDEA poses two problems:
- eslint ignores the setting `ignorePatterns` from `.eslintrc` when running in IDEA.
- IDEA warns about `TODO` in dist/index.js.

Both can be suppressed through marking the `dist` directory as `Excluded`. But to make that work effectively some .idea settings must be kept.

**Comment by MikeN123** [comment]
To be honest, we update this action once every 3 years, so I don't really care too much about what IntelliJ thinks...

**Comment by MikeN123** [comment]
@johanwortel was this PR ready to merge?

**Comment by johanwortel** [comment]
> @johanwortel was this PR ready to merge?

Now it is. It has been tested with a PR in api-echo, with and without a fixup commit.
There are version tags, but strangely enough the action is always used with the master branch qualifier.

**Comment by MikeN123** [comment]
Guess noone bothered to actually use the versions...

**Review by MikeN123 — APPROVED** [review]

**Pull request merged**